Adjusted alias doc for clarity (#120437)
elastic/elasticsearch: docs/reference/alias.asciidoc

[[aliases]]
= Aliases

An alias points to one or more indices or data streams. Most {es}
APIs accept an alias in place of a data stream or index name.

> Using aliases in your application's {es} requests allows you to update or change backing data with no downtime or code changes. This allows production systems to instantly switch between data streams or indices of an alias, without interruption.

Reviewer: ```suggestion
Aliases enable you to:

* Query multiple indices/data streams together with a single name
* Change which indices/data streams your application uses in realtime
* <<docs-reindex,Reindex>> data without downtime
```

Sorry initial edits were a bit rushed, but you've removed the explicit mention of reindexing which is a key use case for aliases. This edit tries to make the 3 major use cases super clear, and the bulleted list keeps verbiage concise. LMKWYT :)

> An alias is a secondary name for a group of data streams or indices. Most {es}

Reviewer: ```suggestion
An alias is a group of indices or data streams that can be referenced by a single name. Most {es}
```

Reviewer: Think we can improve the top-level definition too :)

Reviewer: technically an alias isn't always a group, e.g. it can just point to a single index

Author: Good catch

APIs accept an alias in place of a data stream or index name.

> If you use aliases in your application's {es} requests instead of directly using indices, you can update or change the backing data with **no

Reviewer 2: ```suggestion
Using aliases in your application's {es} requests allows you to update or change backing data with no downtime or code changes. This allows production systems to instantly switch between data streams or indices of an alias, without interruption.
```

Reviewer: (I edited my edit, it's getting very meta here)

Reviewer: Reindexing is key use case for aliases
